### PR TITLE
Add support for some escapes

### LINKED ISSUE #66: Support escapes in our strings.
Our strings do not support escapes like `\t`, `\n`.
`print("Foo \nBar")` now gives 
```
Foo \nBar
```
it should give
```
Foo
Bar
```

### PR DESCRIPTION
**Pull request by XFFXFF**
fix #66

### CODE REVIEW

File: components/dada-lex/src/lex.rs [hunk 1]
```diff
@@ -203,6 +215,19 @@ where
                 break;
             }
 
+            if ch == '\\' {
```
Comment (nikomatsakis) [inline components/dada-lex/src/lex.rs:218]: I think this should be done in a later phase, after lexing-- in the parser or during validation, for example. In particular, I think that explicitly written `\n` and `\t` etc should never be stripped by the whitespace normalization code.

Comment (XFFXFF) [reply components/dada-lex/src/lex.rs:218]: Got it, I'll fix it later.

File: dada_tests/interpret/string-escape.dada [hunk 2]
```diff
@@ -0,0 +1,21 @@
+async fn main() {
+    print("Foo\nBar").await
+
+    print("Foo\tBar").await
+
+    print("Foo\r\nBar").await
+
+    print("Foo \\ Bar").await
+
+    print("Foo \" Bar").await
```
[the remark this answers is not in the document]
Comment (XFFXFF) [reply dada_tests/interpret/string-escape.dada:10]: Should we trim the explicitly written `\n` and `\t` at the start of a line?
```
print("
\tFoo
\tBar
")
```
should give
```
Foo
Bar
```
or
```
    Foo
    Bar
```

File: components/dada-lex/src/lex.rs (hunk 1 above)
Comment (XFFXFF) [reply components/dada-lex/src/lex.rs]: I modified it, feel free to request changes.

**Comment by XFFXFF** [comment]
I removed the test case `"Foo \" Bar"`， because the lexer now thinks `"Foo \"` is a string literal. That's another bug, I'll open an issue later.
https://github.com/dada-lang/dada/blob/ecd3aab2a960d689b22c65b00d4009cc26611791/components/dada-lex/src/lex.rs#L197-L205

File: dada_tests/interpret/string-escape.dada (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikomatsakis) [reply dada_tests/interpret/string-escape.dada]: I think it should give `\tFoo\tBar` -- i.e., we trim the newlines, but we keep the stuff written with `\t`

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**